### ISSUE TITLE
allow cheat files to have extension

### ISSUE THREAD

**Issue by guoqiao** [cheat/cheat#625]
**Is your feature request related to a problem? Please describe.**

Currently the cheat files can not have extension, this caused 2 problems:

1. although optional, I need to specify special heads in file, e.g.:
```
---
syntax: javascript
tags: [ array, map ]
---
```
This is a bit memory burden, and make cheat files not generic.

2.  without extension, syntax highlight won't work directly in editors.

I like to use markdown format in vscode.
However, when I open a cheat file such as `mysql` with vscode, It can not detect the correct type.
This also applies to bash, javascript  syntax types.

**Describe the solution you'd like**

I would like this tool can support file extensions.

For example, if I want to create a cheat file with markdown syntax, I just save it as `tcpdump.md`.
This is a valid/generic markdown file,  which means users can write a cheat file without learning anything.
In file top, I don't need to begin with the special header.
And, when I edit in editor, I get correct syntax highlighting.

What's more interesting,  I can also create cheat files for other syntax, such as `requests.py`, `awk.sh`, `jquery.js`.
I can even run it to verify it's working, although that may leads to some problems, such as you have `sudo rm /` in your script.
But there are many ways to avoid such things.

From the cheat code, that means, `cheat  mysql`  will not only search for a file with exact name `mysql`, it will try to match any files with basename `mysql`, such as `mysql.txt`, `mysql.sh`, `mysql.md`, and `mysql`.
That change won't be too hard, and when there are multiple ones, we can ask user to choose which version.

**Describe alternatives you've considered**

**Additional context**

**Comment by rickswe** [comment]
I agree. Same thing with neovim I have to put this at the end of every file:

```sh
# vim:ft=conf
```

I can create an `autocmd` but don't want to do this for something simple. If anyone has
a better solution please let me know.
